### ISSUE TITLE
Using asNexusMethod for datetime / json (graphql types)

### ISSUE THREAD

**Issue by TommyCoding** [graphql-nexus/schema#565]
I am trying to implement the types for date(time)/json as suggested in the docs (https://nexusjs.org/docs/api/scalar-type) but for my case they or at least type checking doesn`t seem to work.

I have a single _shared.ts_ file with the following code:

```
import { interfaceType } from '@nexus/schema'

import { asNexusMethod } from '@nexus/schema'
import { DateTimeResolver, JSONObjectResolver } from 'graphql-scalars'

export const jsonScalar = asNexusMethod(JSONObjectResolver, 'json')
export const dateTimeScalar = asNexusMethod(DateTimeResolver, 'date')

export const SoftDelte = interfaceType({
    name: 'SoftDelete',
    definition(t) {
      t.date('createdAt') // VSCode reports that "date" is unknown
      t.date('updatedAt') // VSCode reports that "date" is unknown
      t.date('deletedAt') // VSCode reports that "date" is unknown

      t.resolveType(() => null)
    },
  })
```

This is my package.json:

`{
  "name": "test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "dev": "ts-node-dev --transpile-only --no-notify api/app.ts",
    "build": "tsc"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@nexus/schema": "^0.16.0",
    "@prisma/client": "^2.9.0",
    "apollo-server": "^2.18.2",
    "apollo-server-express": "^2.18.2",
    "express": "^4.17.1",
    "graphql": "^15.3.0",
    "graphql-scalars": "^1.4.0",
    "nexus-plugin-prisma": "^0.21.0"
  },
  "devDependencies": {
    "@prisma/cli": "^2.9.0",
    "ts-node-dev": "^1.0.0-pre.65",
    "typescript": "^4.0.3"
  }
}
`

I am running `npm run dev` while in the code window.

Please help!

**Comment by Sytten** [comment]
So you have two options:
1. Use the backing types maps
```typescript
makeSchema({
    ...
    typegenAutoConfig: {
        backingTypeMap: {
              DateTime: 'Date',
        }
    }
})
```
2. Use the root typing param
```typescript
asNexusMethod(DateTimeResolver, 'date', 'Date')
```

Now this works well for built-in types, but if you have types that needs to be imported (say `JsonObject` from [type-fest](https://github.com/sindresorhus/type-fest/)), you need something more fancy. I am doing this on top of my head, but it should look something like
1. 
```typescript
makeSchema({
    ...
    typegenAutoConfig: {
        backingTypeMap: {
              JSONObject: 'typefest.JsonObject',
        },
        sources: [
              {
                  alias: 'typefest',
                  source: 'type-fest',
              },
        ]
    }
})
```
2.
```typescript
asNexusMethod(JSONObjectResolver, 'json', {
    path: 'typefest',
    name: 'JsonObject'
    // Might need the alias here, not sure
})
```

@Weakky @jasonkuhrt  This is a good one for documentation

**Comment by TommyCoding** [comment]
For **date** if I only use option 1) it doesn`t work I also need to add option 2) to have the t.date() available or the following error will come up:

> [ERROR] 11:31:17 TypeError: t.date is not a function

For **Json** it doesn`t work (I installed the package and typescript is at 4.0). The following error comes up:

> GraphQL Nexus: Unable to find file or module type-fest, skipping
> GraphQL service ready at http://localhost:4000/graphql
> Missing required typegen import: typefest

Btw and what if I use the prisma plugin how to configure scalars then (if needed) ?

**Comment by Sytten** [comment]
Honestly "it doesn't work" is just not good enough to help debug and it is also rude, it **does** work because I have two projects in production that uses that technique. Option 1 is in complement of the `asNexusMethod`.
I have no idea for prisma plugin since I don't use it.

**Comment by jasonkuhrt** [comment]
Hey everyone, I don't think anyone is being rude here, all good. Let's relax and keep an welcoming atmosphere :)

Thanks @Sytten looks like some good doc content here indeed.

@TommyCoding About your problems with `json` typing, if you can share a repo repro with us that would help a lot.

**Comment by tgriesser** [comment]
I believe this is fixed in 0.19+
